front/app/containers/Admin/projects/projectFolders/components/ProjectFolderRow/ProjectFolderRow.test.tsx

```diff
@@ -58,20 +76,41 @@ describe('ProjectFolderRow', () => {
   });
 
   describe('When user is a folder moderator', () => {
-    // it('shows the edit button for a folder the user is a moderator of', () => {
-    //   render(<ProjectFolderRow {...props} />);
-    //   const editButton = screen.getByRole('button', { name: 'Edit' });
-    //   expect(editButton).toBeInTheDocument();
-    // });
-    // it('shows a disabled edit button for a folder the user is not a moderator of', () => {
-    //   render(<ProjectFolderRow {...props} />);
-    //   const editButton = screen.getByRole('button', { name: 'Edit' });
-    //   expect(editButton).toBeDisabled();
-    // });
-    // it('does not show the MoreActionsMenu', () => {
-    //   render(<ProjectFolderRow {...props} />);
-    //   const moreActionsMenu = screen.queryByTestId('folderMoreActionsMenu');
-    //   expect(moreActionsMenu).toBeNull();
-    // });
+    beforeAll(() => {
+      mockUserData.attributes.highest_role = 'project_moderator';
+    });
+
+    it('shows the edit button for a folder the user is a moderator of', () => {
+      mockUserData.attributes.roles = [
+        {
+          type: 'project_folder_moderator',
+          project_folder_id: 'publicationId',
+        },
+      ];
+
+      render(<ProjectFolderRow {...props} />);
+      const editButton = screen.getByRole('button', { name: 'Manage' });
+      expect(editButton).toBeInTheDocument();
+    });
+
+    it('shows a disabled edit button for a folder the user is not a moderator of', () => {
+      mockUserData.attributes.roles = [
+        { type: 'project_folder_moderator', project_folder_id: 'testId' },
+      ];
+
+      render(<ProjectFolderRow {...props} />);
+      const editButton = screen.getByTestId('edit-button');
```

Comment:
Since we are already checking using getByRole some where in here and we have more than one instance because because of multiple rendering, the specificity here should be good.

Reply:
You could make the byRole specific by adding the button's title (like we do in other tests of this this file). :-)

Reply:
In this case, we need the specificity because we have multiple renders. We could look into that but I'm not sure it gives a big advantage in this case

Reply:
> In this case, we need the specificity because we have multiple renders. We could look into that but I'm not sure it gives us a big advantage in this case since we are already testing byRole in the file

I'm not sure if I understand what you mean with "we need the specificity because we have multiple renders".

In any case, looking into this then I discovered some (potential) issues:
1. We're grabbing the div not the button if we use data-testid.  
2. Not sure how exactly, but as a result toBeDisabled doesn't work well, so we use toHaveAttribute.
3. If I replace your instance of byTestId with byRole, I have a problem in this test because it also catches the Edit button of the project row in the folder row. This means the tests above (which use byRole) may wrongly pass (if we grabbed the edit button of the project row inside the folder row).

Could you still check if this can generate false positives?

Reply:
@EdwinKato No need to give further explanations of your reasoning, that's too time-consuming. If we still need to clarify things, let's call via Tuple after the standup.